modules/dailymotionBidAdapter.js

```diff
@@ -1,11 +1,43 @@
 import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { ortbConverter } from '../libraries/ortbConverter/converter.js';
 import { VIDEO } from '../src/mediaTypes.js';
 import { deepAccess } from '../src/utils.js';
 import { config } from '../src/config.js';
 import { userSync } from '../src/userSync.js';
 
 const DAILYMOTION_VENDOR_ID = 573;
 
+const dailymotionOrtbConverter = ortbConverter({
+  context: {
+    netRevenue: true,
+    ttl: 600,
+  },
+  imp(buildImp, bidRequest, context) {
+    const imp = buildImp(bidRequest, context);
+
+    if (typeof bidRequest.getFloor === 'function') {
+      const size = imp.w > 0 && imp.h > 0 ? [imp.w, imp.h] : '*';
+
+      const floorInfo = bidRequest.getFloor({
+        currency: 'USD',
+        mediaType: 'video', // or '*' for all the mediaType
+        size
+      }) || {};
+
+      if (floorInfo.floor && floorInfo.currency) {
+        imp.bidfloor = floorInfo.floor;
+        imp.bidfloorcur = floorInfo.currency;
+      }
+    }
+
+    return imp;
+  },
+});
+
+function isArrayEmpty (_array) {
+  return !_array || (Array.isArray(_array) && _array.length === 0)
```

Comment:
Since we are only using this function to test for non-empty arrays, could we use it to validate that the object is indeed an array and it's not empty ?
Ex:
```
function isArrayFilled (_array) {
  return _array && Array.isArray(_array) && _array.length !== 0
}
```
The current function will return false if the object is not an array, and in this case we'll map `iabcat1` or `iabcat2` to something that's not an array, which may cause issues later.

Reply:
Indeed, nice catch 👍